charmcraft-24.04.yaml

```
@@ -33,6 +33,8 @@ extra-bindings:
 requires:
   consul-cluster:
     interface: consul-cluster
+  consul-notify:
+    interface: consul-notify
   juju-info:
     interface: juju-info
     scope: container
```

What's your thoughts about my previous comment on the direction of the relation?

To explain my thought process, it's the consul-client operator that will provide the notifications to the hypervisor charm.


This means, consul-notify should be moved to the `provides:` section instead of requires.

Moreover, this relation encapsulates the reason why we use `juju-info`, I believe `juju-info` should go away, and the scope `container` should be applied to consul-notify relation.

juju-info is a default relation always present that's a good candidate for subordinate when there's no exchange between 2 applications. Now, `consul-notify` plays this role, that's why it's enough to replace `juju-info`

```suggestion
requires:
  consul-cluster:
    interface: consul-cluster

provides:
  consul-notify:
    interface: consul-notify
    scope: container
```

---

+1 for consul-client-operator to be consul-notify provider and openstack-hypervisor to be requirer.

We might still require `juju-info` relation because the idea is not to integrate all consul-client instances to openstack-hypervisor because once they are integrated any NIC down case will lead to shutting down the VMs by openstack-hypervisor snap. The integration should be based on matrix.yaml set in masakari-monitors, consul-client-<> need to be integrated only for the cases recovery action is configured in masakari-monitors.

Here is how it should be decided
1. Deployment has separate networks for management, tenant and storage then only consul-client-storage should be integrated to openstack-hypervisor.
2. Deployment has 2 networks, then see if storage network is defined. If storage network is defined, consul-client-storage should be integrated to openstack-hypervisor. if not, management network is considered as default network for storage, so consul-client-management should be integrated to openstack-hypervisor
3. Deployment has single network, then consul-client-management should be integrated to openstack-hypervisor

I haven't looked at the complete code, i will review once the changes are done in consul-notify